### PR TITLE
Organizations for Country Page

### PR DESCRIPTION
**Pull request by Christoph98**
-Added organizations schema
-Added organizations into country schema
-Added getOrganizations to country resCodes
-Implemented get-organizations API

### CODE REVIEW

File: code/models/country.js [hunk 1]
```diff
@@ -11,6 +13,7 @@ const countrySchema = new mongoose.Schema({
     map: String, // Same as above 
     locationID: { type:mongoose.Schema.Types.ObjectId, ref: 'Location' }
   },
+  organizations: [OrganizationSchema]
```
Comment (thyo9470) [inline code/models/country.js:16]: This isn't how foreign keys are assigned. Check out the trip.js model's locationID. Also, change the name to organizationIDs

File: code/models/organization.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+// Source: https://stackoverflow.com/questions/40102372/find-one-or-create-with-mongoose
+OrganizationSchema.statics.findOneOrCreate = function findOneOrCreate(condition, callback) {
+  const self = this
+  self.findOne(condition, function(err, result){
+    return result ? callback(err, result) : self.create(condition, (err, result) => { return callback(err, result) })
+  });
+}
+
+const organization = mongoose.model('Organization', organizationSchema);
```
Comment (thyo9470) [inline code/models/organization.js:18]: In the database update I am doing I am changing these variables names to be all uppercase. please change to Organization.

**Comment by Christoph98** [comment]
Okay, changes have been made.

File: code/models/country.js [hunk 3]
```diff
@@ -1,15 +1,18 @@
 
 const mongoose = require('mongoose');
 const crypto = require('crypto');
+const Organization = require('./organization');
+const OrganizationSchema = mongoose.model('Organization').schema;
 
 const countrySchema = new mongoose.Schema({
   name: { type: String, required: true },
   abbreviation: { type: String, required: true },
+  organizationID: [{type: mongoose.Schema.Types.ObjectId, ref: 'Organization'}],
```
Comment (thyo9470) [inline code/models/country.js:10]: please make plural ( organizationIDs ) since this is an array

**Comment by natnaelmesfun** [comment]
Everything else looks good.

**Pull request merged**